drivers/gpio/gpio_aw9523b.c

```diff
@@ -0,0 +1,474 @@
+
+#define AW9523B_GPOMD             BIT(4)
+#define AW9523B_RESET_PULSE_WIDTH 20
+
+enum read_write_toggle_t {
+	READ,
+	WRITE,
+	TOGGLE,
+};
+
+struct gpio_aw9523b_config {
+	struct gpio_driver_config common;
+	const struct device *mfd_dev;
+	struct i2c_dt_spec i2c;
+	bool port0_push_pull;
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(reset_gpios)
+	struct gpio_dt_spec reset_gpio;
+#endif
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+	struct gpio_dt_spec int_gpio;
+	gpio_callback_handler_t int_cb;
+#endif
+};
+
+struct gpio_aw9523b_data {
+	struct gpio_driver_data common;
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+	const struct device *dev;
+	sys_slist_t callbacks;
+	struct gpio_callback gpio_callback;
+	struct k_work intr_worker;
+	gpio_port_value_t prev_value;
+	gpio_port_pins_t rising_event_pins;
+	gpio_port_pins_t falling_event_pins;
+#endif
+};
+
+static int gpio_aw9523b_pin_configure(const struct device *dev, gpio_pin_t pin, gpio_flags_t flags)
+{
+	const struct gpio_aw9523b_config *const config = dev->config;
+	const uint8_t port = (pin < 8) ? 0 : 1;
+	const uint8_t mask = BIT(pin % 8);
+	const uint8_t input_en = (flags & GPIO_INPUT) ? mask : 0x00;
+	const uint8_t out_high = (flags & GPIO_OUTPUT_INIT_HIGH) ? mask : 0x00;
+	int err;
+
+	/* Can't do I2C operations from an ISR */
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+
+	/* Either INPUT or OUTPUT must be set */
+	if ((!!(flags & GPIO_INPUT) ^ !!(flags & GPIO_OUTPUT)) == 0) {
+		return -ENOTSUP;
+	}
+
+	/* Open-drain support is per port, not per pin.
+	 * So can't really support the API as-is.
+	 */
+	if (port == 0 && !config->port0_push_pull) {
+		if (!((flags & GPIO_SINGLE_ENDED) && (flags & GPIO_LINE_OPEN_DRAIN))) {
+			return -ENOTSUP;
+		}
+	} else {
+		if (flags & GPIO_SINGLE_ENDED) {
+			return -ENOTSUP;
+		}
+	}
+
+	if ((flags & GPIO_INPUT) && ((flags & GPIO_PULL_UP) || (flags & GPIO_PULL_DOWN))) {
+		return -ENOTSUP;
+	}
+
+	k_sem_take(aw9523b_get_lock(config->mfd_dev), K_FOREVER);
+
+	err = i2c_reg_update_byte_dt(&config->i2c, AW9523B_REG_CONFIG0 + port, mask, input_en);
+	if (err) {
+		LOG_ERR("%s: Failed to set pin%d direction (%d)", dev->name, pin, err);
+		goto on_error;
+	}
+
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+	if (config->int_gpio.port) {
+		if (input_en) {
+			struct gpio_aw9523b_data *const data = dev->data;
+			uint8_t buf[2];
+
+			/* Read initial pin state */
+			err = i2c_burst_read_dt(&config->i2c, AW9523B_REG_INPUT0, buf, 2);
+			if (err) {
+				LOG_ERR("%s: Read initial pin state failed (%d)", dev->name, err);
+				goto on_error;
+			}
+
+			WRITE_BIT(data->prev_value, pin, !!((buf[1] << 8 | buf[0]) & BIT(pin)));
+		} else {
+			struct gpio_aw9523b_data *const data = dev->data;
+
+			WRITE_BIT(data->falling_event_pins, pin, 0);
+			WRITE_BIT(data->rising_event_pins, pin, 0);
+		}
+	}
+#endif
+
+	err = i2c_reg_update_byte_dt(&config->i2c, AW9523B_REG_OUTPUT0 + port, mask, out_high);
```

Comment:
I'd do a define for `AW9523B_REG_OUTPUT0 + port` so you can do `AW9523B_REG_OUTPUT(port)`

Reply:
I fixed it.